api/v1alpha1/event.go

```diff
@@ -67,15 +67,26 @@ func FormatEventUserActor(u user.Info) string {
 	switch {
 	case u.IsAdmin:
 		return EventActorAdmin
-	case u.Email != "":
-		return EventActorEmailPrefix + u.Email
-	case u.Subject != "":
-		return EventActorSubjectPrefix + u.Subject
+	case IsClaimValueNonEmptyString(u.Claims["emails"]):
+		claimString, _ := u.Claims["emails"].(string)
+		return EventActorEmailPrefix + claimString
+	case IsClaimValueNonEmptyString(u.Claims["subs"]):
+		claimString, _ := u.Claims["subs"].(string)
+		return EventActorSubjectPrefix + claimString
 	default:
 		return EventActorUnknown
 	}
 }
 
+func IsClaimValueNonEmptyString(claimValue any) bool {
+	if claimString, ok := claimValue.(string); ok {
+		if claimString != "" {
+			return true
+		}
+	}
+	return false
+}
```

Comment:
This looks looks like it might be more straightforward:

```golang
func FormatEventUserActor(u user.Info) string {
	email := u.Claims["emails"].(string) // nolint: forcetypeassert
	subject := u.Claims["subs"].(string) // nolint: forcetypeassert
	switch {
	case u.IsAdmin:
		return EventActorAdmin
	case email != "":
		return EventActorEmailPrefix + email
	case subject != "":
		return EventActorSubjectPrefix + subject
	default:
		return EventActorUnknown
	}
}
```

Reply:
Yes thats much better, fixed.